### ISSUE TITLE
Check in

### ISSUE THREAD

**Issue by svenfuchs** [rubycorns/rorganize.it#457]
One big thing that sort of keeps me (us?) using rorganize.it more actively is that keeping track of things is rather tedious. E.g. memberships and actual participation is never really reflected on https://rorganize.it/groups/rubymonsters ... that's kind of a pity.

I think it would be helpful to have kind of a "check in" feature. Participants and coaches would be able to "check in" during the meeting, and select from a list who's currently attending. Others could add on to this. That way our data would probably still never be perfectly up to date, but still much more useful.

"active/inactive" tags could be based on this data, and the membership list could be ordered/tagged based on it, too.

Another big hurdle is that memberships currently require signing up. That's not very practical, especially for newcomers. It would be cool if we could create profiles for others, just using their name and email address, and check them in, e.g. at the first meeting at a new beginners group. They could then later "claim" their profile, if they want to.

**Comment by lislis** [comment]
Thanks for the feedback Sven!

A  few comments/ thoughts from my side:
In your scenario the check-ins would be based on actual meetings of the group, right? We don't have anything like that currently and I'm not sure how to best go about it. Give any member in the group the possibility to create a new meeting that the others can join? Only the coach? Would this be something people are interested? Maybe even attaching topics to meetings? Also, a group's (in)activity would then much rather be based on the last meeting (eg last_meeting.date was over two months ago or something) rather than the check ins. Does that make sense?

I personally am not a big fan of creating profiles for other people. The steps of verifying a user is who they claim to be could turn out quite tricky. Also just randomly creating users and checking others in rings quite a few privacy bells in my head. 

What do other people think of both topics?

**Comment by svenfuchs** [comment]
@lislis ohai :)

hmmm, i didn't have creating meetings as a first class thing in mind when i suggested this. i was more thinking of a "ping" feature ... so that it would be extremely easy and low-cost for people to tell rorganize.it there's been activity. that could be as simple as a big button "check in" which can be used by any group member during an actual meeting. it would simply record these checkins.

thinking more about this i think it would be beneficial if this worked whether or not there's a meeting created on the side of rorganize.it. I can see how "meetings" might be a nice feature (brings you closer to meetup.com) ... but that's also a little extra overhead, so i wouldn't rely on people to do that.

some parts of our groups actually also meet outside the normal meeting slot. e.g. some subgroups meet at home for homework sessions. or they would meet at a code & cake event to continue working on their curriculum. if they could just "check in" and record/display their activity i feel this might give a more realistic impression of how active a group is already.

i understand the concerns about privacy. i guess from a product perspective, if it's hard to make sure the membership data is complete and up to date, then it's also harder to make the application really useful in this specific regard. The current state of the Ruby Monstas details page seems horribly outdated, which kinda gives a bad impression and makes me feel embarrassed about it. I'd therefor tend to click away, since I don't see a way for me to fix that (it probably also would be tons of work even if i was able to manage people manually ... there are just too many folks).

:)

**Comment by lislis** [comment]
> it would simply record these checkins
I don't think that checkins on their own make much sense. They lack a context. If you say it's a `ping` that means something to developers, sure. But beginners would struggle with that concept I think. I'm also thinking of how we would display that in the UI. Like `You checked in 3 weeks ago. <button>Checkin now<button>` seems strange to me when I don't really know what I'm checking into. But maybe that is just me.

> "meetings" might be a nice feature (brings you closer to meetup.com) ... but that's also a little extra overhead
@PragTob brought up that we could create meetings for groups automatically based on their meeting schedule. Then groups can delete them if it didn't happen or they're automatically deleted if nobody checked into the meeting after a week or so. Plus groups can create custom meetings.

**Comment by svenfuchs** [comment]
Of course! This sounds like it would be way more work for you, but yeah, sounds like this would cover our usecase, too.

:heart:
